packages/framework/src/client.ts

```diff
@@ -829,6 +829,19 @@ export class Client {
     return getCodeResult;
   }
 }
+
+type ChannelStepOption = {
```

Comment:
the option type here is not correct and is hard to reuse because of its dynamic nature; therefore, i created a simple type for now.
in addition I created a [ticket](https://linear.app/novu/issue/NV-5508/fix-return-value-of-executestepfactory-in-novuframework) to solve the typing issue.